### PR TITLE
Remove unnecessary Arc and fix typo

### PR DESCRIPTION
**Pull request by cecton**

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]

**Review by Geal — APPROVED** [review]

File: apollo-router-core/src/traits.rs [hunk 1]
```diff
@@ -70,11 +70,11 @@ where
 
 impl<T: ?Sized> WithCaching for T where T: QueryPlanner + Sized + 'static {}
 
-/// An object that accepts a [`Request`] and allow creating [`PreparedQuery`]'s.
+/// An object that accepts a [`Request`] and allows creating [`PreparedQuery`]'s.
```
Comment (garypen) [inline apollo-router-core/src/traits.rs:73]: NIT: Slightly better grammar to use "allows for".

File: apollo-router-core/src/traits.rs [hunk 2]
```diff
@@ -70,11 +70,11 @@ where
...
 ///
-/// The call to the function will either succeeds and return a [`PreparedQuery`] or it will fail and return
-/// a [`Response`] that can be returned immediately to the user. This is because GraphQL does
-/// not use the HTTP error codes, therefore it always return a response even if it fails.
+/// The call to the function will either succeeds and return a [`PreparedQuery`] or it will fail
```
Comment (garypen) [inline apollo-router-core/src/traits.rs:75]: NIT: Might as well fix "succeeds" to "succeed" whilst we are fixing the other text.

**Review by garypen — COMMENTED** [review]
A couple of text NITS...

File: apollo-router-core/src/traits.rs (hunk 1 above)
Comment (cecton) [reply apollo-router-core/src/traits.rs:73]:
```suggestion
/// An object that accepts a [`Request`] and allows for creating [`PreparedQuery`]'s.
```

**Pull request merged**